### ISSUE TITLE
teyoのマージ前にやってきたこと

### ISSUE THREAD

**Issue by teyosan** [trtd56/BirdCLEF#36]
## リソース

+ kaggleのGPU/TPU
+ colabpro
+ ローカルマシン(GTX1070)
  + 基本これでモデルを作ってる　15h/1fold

## 方針

全データ使用、CNNモデルでclassificationのアプローチ

## 0.62でやってたこと

[Code](https://www.kaggle.com/teyosan1229/birdclef-cnnmodel)

+ trainsoundscape
  + f1:0.6718
  + prec:0.6767
  + rec:0.6697
+ valdata
  + microf1:0.7697
  + samplef1:0.6679
  + valloss:0.0043

optimizer,loss,schedulerは↑の通り

+ model
  + rexnet_200
+ input channel 1
  + 今は3に変えてるけど特に変化が起きてない
+ augment
  + noiseinjection
  + randomvolume
  + normalize
  + specaugment
+ mixup
  + [鳥1の3位の手法](https://github.com/TheoViel/kaggle_birdcall_identification/blob/2de708b9871cf388f91b9b0a33e738a24cca565d/src/training/train.py#L15)
  + 効果はあった
+ inference
  + [Code](https://www.kaggle.com/teyosan1229/birdclef-inference-3ch)
  + 鳥1,3位のvotingを使用
  + 前後の窓のscoreも集約して確率の上位を答えとする

## not worked

+ labelsmoothing
+ 鳥1,2位のpowerを0.5から3に変更する処理
+ 3chをlogmel,delta,deltadeltaにする方法
  + 悪化もしないので1chよりはましな気がするので採用したまま
+ GaussianNoise
+ efficientnet
  + 鳥1といいなぜかefficientnetのスコアが出ない・・・
  + effnetv2も＞＜

## 現状

+ colabとkaggleは制限時間もあるため、logmel化されたデータを使いwaveformの処理以外の実験をしようとしてるけど、localと同じような学習を実現できてなくて困ってる・・・batchサイズとlrが結構シビア・・・？
  + colabとkaggleで代わりに実験回せます。
+ 鳥1の3位や4位のコードを結構参考にして進めてる(https://www.kaggle.com/vladimirsydor/4-th-place-solution-inference-and-training-tips?scriptVersionId=42796948)

## 今後試したいこと

+ HLの活用
+ getitemの工夫
  + getitemでnocallっぽかったら上書きするやつ
+ 2層dropout
+ augmentのnoise追加
